src/lightning/fabric/wrappers.py

```diff
@@ -218,15 +219,35 @@ def _unwrap_objects(collection: Any) -> Any:
     def _unwrap(
         obj: Union[_FabricModule, _FabricOptimizer, _FabricDataLoader]
     ) -> Union[nn.Module, Optimizer, DataLoader]:
-        if isinstance(obj, _FabricModule):
-            return obj._forward_module
+        if isinstance(_unwrap_compiled(obj), _FabricModule):
+            return obj._forward_module  # type: ignore[union-attr]
```

Comment:
It's fair that mypy complained here. It doesn't know that if `_unwrap_compiled(obj)` is a `FabricModule`, then `obj._forward_module` exists because the `OptimizedModule` forwards calls to its inner module.

It would be cleaner to assign the unwrapped object and us that one for the instance and return statement. That should remove the need to ignore

Reply:
I don't understand your proposal, what to do

Reply:
```suggestion
        if isinstance(unwrapped := _unwrap_compiled(obj), _FabricModule):
            return unwrapped._forward_module
```